apps/web/pages/api/sendDailyVideoRecordingEmail.ts

```diff
@@ -0,0 +1,137 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { z } from "zod";
+
+import { getDownloadLinkOfCalVideoByRecordingId } from "@calcom/core/videoClient";
+import { sendDailyVideoRecordingEmails } from "@calcom/emails";
+import { defaultHandler } from "@calcom/lib/server";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import prisma, { bookingMinimalSelect } from "@calcom/prisma";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+
+const schema = z.object({
+  recordingId: z.string(),
+  bookingUID: z.string(),
+});
+
+const downloadLinkSchema = z.object({
+  download_link: z.string(),
+});
+
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (!process.env.SENDGRID_API_KEY || !process.env.SENDGRID_EMAIL) {
+    return res.status(405).json({ message: "No SendGrid API key or email" });
+  }
+  const response = schema.safeParse(JSON.parse(req.body));
+  console.log("REQ", response);
+
+  if (!response.success) {
+    return res.status(400).send({
+      message: "Invalid Payload",
+    });
+  }
+
+  const { recordingId, bookingUID } = response.data;
+
+  try {
+    const booking = await prisma.booking.findFirst({
+      where: {
+        uid: bookingUID,
+      },
+      select: {
+        ...bookingMinimalSelect,
+        uid: true,
+        location: true,
+        isRecorded: true,
+        user: {
+          select: {
+            id: true,
+            credentials: true,
+            timeZone: true,
+            email: true,
+            name: true,
+            locale: true,
+            destinationCalendar: true,
+          },
+        },
+      },
+    });
+
+    if (!booking || booking.location !== "integrations:daily") {
+      return res.status(404).send({
+        message: `Booking of uid ${bookingUID} does not exist or does not contain daily video as location`,
+      });
+    }
+
+    const response = await getDownloadLinkOfCalVideoByRecordingId(recordingId);
+
+    const downloadLinkResponse = downloadLinkSchema.parse(response);
+    const downloadLink = downloadLinkResponse.download_link;
+
+    await prisma.booking.update({
+      where: {
+        uid: booking.uid,
+      },
+      data: {
+        isRecorded: true,
+      },
+    });
+
+    const checkMembership = await prisma.membership.findFirst({
```

Comment:
We should aim to reuse useHasPaidPlan logic. I am not sure how or if you can use trpc fn in non-react scope but we can always abstract the logic out and create hasPaidPlan method that can be used here and[ also in trpc query.](https://github.com/calcom/cal.com/blob/d6e3f2fe91668b52e2b2b86888f2963c5e253b61/packages/trpc/server/routers/viewer/teams.tsx#L723)